buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchDistribution.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.gradle.api.Buildable;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.TaskDependency;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.Locale;
+
+public class ElasticsearchDistribution implements Buildable {
+
+    public enum Platform {
+        LINUX,
+        WINDOWS,
+        DARWIN;
+
+        @Override
+        public String toString() {
+            return super.toString().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    public enum Type {
+        INTEG_TEST_ZIP,
```

Comment:
I don't like that because then specifying it in build.gradle files requires doing an import. As it is now, all these values can be set by string.

Reply:
@rjernst 
Not if we make that the default :) Or use some Groovy magic to avoid the import ? 
It's not a big deal, but calling it a type feels wrong  and I think it forces us to do more conditionals down the road.

Reply:
One option is to provide a convenience method on the `ElasticsearchDistribution` container similar to somthing like `mavenCentral()` on `RepositoryHandler`.

```
elasticsearch_distributions {
    integTest()
}```

It just now also occurred to me that we use snake case for this extension name. Thoughts on making this camel case as is convention with most Gradle plugin extensions?